Formula/kyuubi.rb

```diff
@@ -0,0 +1,53 @@
+class Kyuubi < Formula
+  desc "Serverless SQL on data warehouses and lakehouses"
+  homepage "https://kyuubi.apache.org/"
+  url "https://www.apache.org/dyn/closer.lua?path=kyuubi/kyuubi-1.6.1-incubating/apache-kyuubi-1.6.1-incubating-bin.tgz"
+  mirror "https://archive.apache.org/dist/kyuubi/kyuubi-1.6.1-incubating/apache-kyuubi-1.6.1-incubating-bin.tgz"
+  version "1.6.1-incubating"
+  sha256 "a64a6e5d71b39b0e494597282493250df1c74714fc2bba5057946926c0a32ba2"
+  license "Apache-2.0"
+  head "https://github.com/apache/kyuubi.git", branch: "master"
+
+  depends_on "apache-spark" => :test
+  depends_on "openjdk@8"
+
+  def install
+    libexec.install Dir["*"]
+
+    setting = {
+      JAVA_HOME:            Language::Java.overridable_java_home_env("1.8")[:JAVA_HOME],
+      SPARK_HOME:           "${SPARK_HOME:-#{Formula["apache-spark"].opt_libexec}}",
+      KYUUBI_HOME:          libexec.to_s,
+      KYUUBI_LOG_DIR:       "${KYUUBI_LOG_DIR:-#{var}/kyuubi/logs}",
+      KYUUBI_PID_DIR:       "${KYUUBI_PID_DIR:-#{var}/kyuubi/pid}",
+      KYUUBI_WORK_DIR_ROOT: "${KYUUBI_WORK_DIR_ROOT:-#{var}/kyuubi/work}",
+    }
+    (bin/"kyuubi").write_env_script libexec/"bin/kyuubi", setting
+    (bin/"kyuubi-beeline").write_env_script libexec/"bin/beeline", setting
+  end
+
+  def post_install
+    %w[logs work pid zk].each do |dir|
+      (var/"kyuubi"/dir).mkpath
+    end
+  end
+
+  test do
+    ENV.append "_JAVA_OPTIONS", "-Duser.home=#{testpath}"
+    ENV.append "_JAVA_OPTIONS", "-Dkyuubi.frontend.bind.host=localhost"
+    ENV.append "_JAVA_OPTIONS", "-Dkyuubi.zookeeper.embedded.data.dir=#{testpath}/zk"
+    ENV.append "_JAVA_OPTIONS", "-Dkyuubi.metrics.reporters=''"
```

Comment:
Will users need to set all these too?

Reply:
not necessary

Reply:
Then why are they required in this block?

Reply:
kyuubi.frontend.bind.host is hardcoded to localhost for convenience.

kyuubi.zookeeper.embedded.data.dir is set to testpath to avoid polluting the one created by a user.

kyuubi.metrics.reporters is disabled for tests, which reports to a local json file by default,

Reply:
Can we remove anything that isn't absolutely required to run the test. All the extra settings are making it difficult to see how the command would work outside the test block.

Reply:
like @SMillerDev said, can we create a simple config in the kyuubi project and included in here for testing? That might be better and easy to maintain.

Reply:
The custom test configs are a subset of the runtime one now. only `localhost` is used for convenience